### PR TITLE
Implement Metamath spec changes 1-7, fixes #7

### LINKED ISSUE #7: Implement spec changes 1-7
Per Norm Megill:

> One of the most important parts of the book is the 4-page Metamath spec pp. 92-95. In particular, almost every verifier was written using only these 4 pages, even by programmers who have little understanding of Metamath at a higher level.

> I see that not filed under "metamath.tex" in my to-do list is a set of proposed spec changes (mainly minor clarifications or restrictions that don't affect much in practice). These will require small changes to the spec, which will hopefully remain 4 pages. :)

Proposed spec changes #1-#7 are at:
https://groups.google.com/d/msg/metamath/ZlRle52FVO0/7TqSetEtCQAJ

### PR DESCRIPTION
**Pull request by david-a-wheeler**
This implements Metamath spec changes 1-7 as documented
in issue #7 and also in
https://groups.google.com/d/msg/metamath/ZlRle52FVO0/7TqSetEtCQAJ

I simplified some other text, and merged a few paragraphs, to
ensure that the spec stays well within 4 PDF pages.

Signed-off-by: David A. Wheeler <dwheeler@dwheeler.com>

### CODE REVIEW

**Comment by david-a-wheeler** [comment]
@nmegill - okay, here's the "real deal" - text that *changes* the Metamath spec per changes 1-7.  You already agreed to the changes conceptually.  Here I try to turn it into actual text in the spec itself.

Comments welcome!

**Comment by nmegill** [comment]
Let me look it over and I'll get back, probably tomorrow.

File: metamath.tex [hunk 1]
```diff
@@ -7374,7 +7374,7 @@ \chapter{The Metamath Language}
 
 The next section contains the complete specification of the Metamath
 language.\footnote{The current version of the Metamath program
-(0.07.30)\index{Metamath!limitations of version 0.07.30} implements an
```
Comment (nmegill) [inline metamath.tex]: Let's take out this entire footnote.  (metamath.exe should implement it exactly unless there is a bug.)

**Comment by david-a-wheeler** [comment]
Sure! Any other changes?

File: metamath.tex [hunk 2]
```diff
@@ -7440,31 +7439,42 @@ \subsection{Preprocessing}
...
-keywords}\index{included file}\index{file inclusion}  The file name may not
-contain a \texttt{\$} or white space.  The file must exist.  The case-sensitivity
+keywords}\index{included file}\index{file inclusion}
+It is only allowed in the outermost scope (i.e., not between 
+\texttt{\$\char`\{} and \texttt{\$\char`\}})
+and must not be inside of a statement (e.g., it may not occur
+between the label of a \texttt{\$a} statement and its \texttt{\$.}).
+The file name may not
+contain a \texttt{\$} or white space.  The file must exist, and is
+relative to the current directory (which might not be the directory
```
Comment (nmegill) [inline metamath.tex]: The problem of directory paths is discussed in the thread of this post:
https://groups.google.com/d/msg/metamath/eI0PE0nPOm0/f8_btXt4AgAJ
In Stefan's post prior to this, he suggests we can "ust use underscores to separate semantic levels" which is what we are doing now.  I'm not sure how to word it, but I think we should discourage directory paths in file names at this point (if not forever).

File: metamath.tex [hunk 3]
```diff
@@ -7550,12 +7564,11 @@ \subsection{Basic Syntax}
...
-
-No label token may match any math symbol
-token.\label{namespace}\footnote{{\em (Added June 24, 2006)} This
-restriction did not exist in earlier versions of this specification.
-While not theoretically necessary, it is imposed to make it easier to
+Each label token must be unique, and
+no label token may match any math symbol
+token.\label{namespace}\footnote{This
+restriction was added on June 24, 2006.
+It is not theoretically necessary, but is imposed to make it easier to
```
Comment (nmegill) [inline metamath.tex]: "It is not theoretically necessary, but is imposed" - take out comma before "but"

**Comment by nmegill** [comment]
I'm all set for now, so it can be merge after the changes I suggest.

A somewhat open item is how to discuss directory paths, but maybe we should just discourage them altogether for now - make them "not supported" or "OS dependent".  As for the definition of "current directory", perhaps we should say that it is typically the directory in which the verifier is launched but may vary with verifier (e.g. it is a parameter in mmj2).

This is the first time I've tried to mark up the diff page with comments, and it seems each comment is creating a new email, making it confusing if I have to revise a comment before I'm done.  Do you know of any way to have all comments sent at once?

**Comment by david-a-wheeler** [comment]
I don't know of a way to have all comments sent at once, but I don't find the current feedback confusing.

Recommendation: Let's handle directories-in-included-filenames as a *completely* separate issue.  I'd like to make forward progress with what we have.

File: metamath.tex (hunk 1 above)
Comment (david-a-wheeler) [reply metamath.tex]: Done, it'll be in the next commit on this branch.

**Pull request merged**